ocaml/idl/datamodel.ml

```diff
@@ -5975,6 +5975,61 @@ module VM_appliance = struct
       ()
 end
 
+module VM_group = struct
+  let placement_type =
+    Enum
+      ( "vm_placement_type"
+      , [
+          ("anti_affinity", "Anti-affinity group")
+        ; ("normal", "Normal type group")
+        ]
+      )
+
+  let create =
+    call ~name:"create" ~in_oss_since:None ~lifecycle:[]
```

Comment:
Does is make sense to have more than one group of the normal type? What are the the semantics of that?

Reply:
As the `normal` is just a default type as a placeholder, user will never need to use this type of placement, is it possible to hide it from api (and cli)?

Reply:
I went through xapi code and failed to find similar placeholder for some type as `normal` here. As it is not required, I wonder is it a required Enum type for the `placement`?

Reply:
I think we have 2 options:
1 Allow users to create `normal` groups and don't limit the group number.
2 We should remove the' normal' type if we don't allow users to create `normal` groups.

Reply:
I understand the `normal` is a default value of the `placement` field of a VM group. By design, the VM group is a more general concept than `VM placement group`. So it is supported to create a VM group without considering the `placement` attribute. In this case, the `normal` is a default value for VM groups which don't care about the `placement`.

Reply:
If the `placement` field is a required parameter when creating a VM group, can we remove the `normal` type and leave only `anti_affinity`? Since what we can see now is that only an anti-affinity group is needed.
We are trying to removing the `normal` as default value for `placement` when creating a VM group, but the build did not pass, will have some more try and update here.

Reply:
How to specify the default value is an implementation detail.
If it is designed as `VM_group` instead of `VM_placement_group`, I understand it has to have a default value (not anti-affinity) for other `VM_group`.

Reply:
> I understand it has to have a default value (not anti-affinity) for other VM_group

I think now we only need `anti_affinity` VM_group, or besides `anti_affinity`, what is the other type of VM group needed?

Reply:
If the design is to come up with a general VM group concept, then we can't enforce there will be only one specific type of group. Otherwise, the design should define a specific group for this purpose.

Reply:
Seems we can not have `placement` without a default value as we can this check here: https://github.com/xapi-project/xen-api/blob/master/ocaml/idl/dm_api.ml#L299-L304. Otherwise we have to change it to `DynamicRO`, but for a VM group, it's `placement` type should not be a variable.
So I think this is a reason why we need a default type like `normal`. @BengangY I think we can either restrict the type not equal to `normal` when creating a VM group or just leave it not handled.(I think there is no harm if more `normal` VM groups are created).